table/two_level_iterator.cc

```diff
@@ -158,14 +158,9 @@ void TwoLevelIterator::Prev() {
 
 
 void TwoLevelIterator::SkipEmptyDataBlocksForward() {
-  while (second_level_iter_.iter() == nullptr ||
```

Comment:
We changed the while to and if condition, looks to me that this is based on the assumption that we dont have empty data blocks, I cannot think of a situation right now where we will have empty data blocks but I dont think that this is a safe assumption. It maybe safe to have this assumption with block based table but I am not sure about other table formats (plain, cuckoo) 

By empty data blocks I mean that after creating the second_level_iter_ for the new block we have second_level_iter_.Valid() == false